infra/Pos/Network/Types.hs

```diff
@@ -119,11 +119,16 @@ topologyNodeType TopologyLightWallet{} = NodeEdge
 
 -- | The NodeType to assign to subscribers. Give Nothing if subscribtion
 -- is not allowed for a node with this topology.
+--
+-- TODO: We allow corf nodes to run Kademlia, but we do not run the subscription
+-- listener on them currently. We may wish to make that configurable.
```

Comment:
Na, I think it's fine. Core nodes running kademlia is plausible, just as running it on relays, but that doesn't mean it uses kademlia for anything, it's just seeding the network. It's independent of allowing subscribers.